### PR TITLE
added  BufferedInputStream and closed the io stream

### PR DESCRIPTION
**Pull request by xiaodaigh**
and minor changes to zip and gzip commands to make them correct

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -307,23 +307,25 @@ using CSV
 CSV.write("c:/data/a.csv", a)
 
 # Windows users who do not have gzip available on the PATH should manually gzip the CSV
-;gzip a.csv
+;gzip c:/data/a.csv
 
-using CodecZlib
-@time a= CSV.read(GzipDecompressorStream(open("c:/data/a.csv.gz")))
+using CodecZlib, BufferedStreams
+io = open("c:/data/a.csv.gz") |> GzipDecompressorStream |> BufferedInputStream
```
Comment (quinnj) [inline docs/src/index.md]: I'd prefer something more idiomatic Julia, like:
```julia
@time a = open("c:/data/a.csv.gz") do io
    CSV.read(GzipDecompressorStream(io))
end
```

Comment (xiaodaigh) [reply docs/src/index.md]: I F up by deleting the branch T_T. Do you think you can finish off the PR?

Comment (nalimilan) [reply docs/src/index.md]: Also I don't understand why you introduce Windows-specific paths here.

Comment (xiaodaigh) [reply docs/src/index.md]: Just an example. Was easier to test as I was writing it

Comment (nalimilan) [reply docs/src/index.md]: Yes but we'd better keep things OS-neutral when possible.

**Comment by xiaodaigh** [comment]
superseded by #520

**Pull request closed without merging**